core/new-gui/src/app/common/service/user/auth.interceptor.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from "@angular/core";
+import { HttpErrorResponse, HttpEvent, HttpHandler, HttpInterceptor, HttpRequest } from "@angular/common/http";
+import { Observable, throwError } from "rxjs";
+import { catchError } from "rxjs/operators";
+import { Router } from "@angular/router";
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  constructor(private router: Router) {}
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    return next.handle(request).pipe(
+      catchError((err: unknown) => {
+        if (err instanceof HttpErrorResponse) {
+          if (err.status === 401) {
```

Comment:
just want to check if we have 403 error code (forbidden) from our endpoints, and a question is do we want to handle 403 the same as 401 here?

Reply:
The JwtConsumer will always return 401 when the key is invalid.
We are checking only 401 code on purpose such that we only redirect when the backend got an invalid JWT key.